utils/src/sync/semaphore.rs

```diff
@@ -15,13 +39,40 @@ use std::{
 pub(crate) struct Semaphore {
     counter: AtomicUsize,
     signal: Event,
+    #[cfg(feature = "semaphore-trace")]
+    readers_start: AtomicU64,
+    #[cfg(feature = "semaphore-trace")]
+    readers_end: AtomicU64,
+    #[cfg(feature = "semaphore-trace")]
+    readers_time: AtomicU64,
+    #[cfg(feature = "semaphore-trace")]
+    log_time: AtomicU64,
+    #[cfg(feature = "semaphore-trace")]
+    log_value: AtomicU64,
 }
 
 impl Semaphore {
     pub const MAX_PERMITS: usize = usize::MAX;
 
     pub const fn new(available_permits: usize) -> Semaphore {
-        Semaphore { counter: AtomicUsize::new(available_permits), signal: Event::new() }
+        cfg_if::cfg_if! {
+            if #[cfg(feature = "semaphore-trace")] {
+                Semaphore {
+                    counter: AtomicUsize::new(available_permits),
+                    signal: Event::new(),
+                    readers_start: AtomicU64::new(0),
+                    readers_end: AtomicU64::new(0),
+                    readers_time: AtomicU64::new(0),
+                    log_time: AtomicU64::new(0),
+                    log_value: AtomicU64::new(0),
+                }
+            } else {
+                Semaphore {
+                    counter: AtomicUsize::new(available_permits),
+                    signal: Event::new(),
+                }
+            }
+        }
```

Comment:
```rust
pub fn new(available_permits: usize) -> Semaphore {
        cfg_if::cfg_if! {
            if #[cfg(feature = "semaphore-trace")] {
                Semaphore {
                    counter: AtomicUsize::new(available_permits),
                    signal: Event::new(),
                    trace_inner: Default::default(),
                }
            } else {
                Semaphore {
                    counter: AtomicUsize::new(available_permits),
                    signal: Event::new(),
                }
            }
        }
    }
```

if previous suggestions accepted, this would be possible